lib/rules/header.js

```diff
@@ -5,7 +5,7 @@ var commentParser = require("../comment-parser");
 var os = require("os");
 
 function isPattern(object) {
-    return typeof object === "object" && object.hasOwnProperty("pattern");
+    return typeof object === "object" && Object.prototype.hasOwnProperty.call(object, "pattern");
```

Comment:
Why is this change needed? If it exist on the prototype it must also exist on the object?

Reply:
This project has this eslint rule enabled https://eslint.org/docs/rules/no-prototype-builtins